packages/core/src/shader/utils/uniformParsers.ts

```diff
@@ -149,6 +149,62 @@ export const uniformParsers: IUniformParser[] = [
                     data[offset+3] = v.height;
                 `
     },
+    // upload a pixi color as vec4 with caching layer
+    {
+        test: (data: any, uniform: any): boolean =>
+            data.type === 'vec4' && data.size === 1 && !data.isArray && uniform.red !== undefined,
```

Comment:
Is there a reason you wouldn't just do `uniform instanceof Color` instead of checking `uniform.red !== undefined`?